File: common-module/src/main/java/ru/kraser/technical_helper/common_module/dto/breakage/UpdateBreakagePriorityDto.java
package ru.kraser.technical_helper.common_module.dto.breakage;

import jakarta.validation.constraints.NotNull;
import ru.kraser.technical_helper.common_module.enums.Priority;
import ru.kraser.technical_helper.common_module.enums.Status;

public record UpdateBreakagePriorityDto(
        @NotNull(message = "Заявка на неисправность должна иметь корректный приоритет.")
        Priority priority,
        @NotNull(message = "Заявка на неисправность должна иметь корректный статус.")
        Status status
) {
}

File: frontend/src/app/model/breakage/update-breakage-priority-dto.ts
import { Priority } from "../../enum/priority.enum";
import { Status } from "../../enum/status.enum";

export class UpdateBreakagePriorityDto {
    priority: Priority;
    status: Status;

    constructor(priority: Priority, status: Status) {
        this.priority = priority;
        this.status = status;
    }
}